src/OpenTelemetry.Instrumentation.AspNet/Implementation/HttpInListener.cs

```diff
@@ -76,7 +76,8 @@ public override void OnStartActivity(Activity activity, object payload)
             {
                 var ctx = this.options.Propagator.Extract(default, request, HttpRequestHeaderValuesGetter);
 
-                if (ctx.ActivityContext.IsValid() && ctx.ActivityContext != activity.Context)
+                if (ctx.ActivityContext.IsValid()
+                    && ctx.ActivityContext != new ActivityContext(activity.TraceId, activity.ParentSpanId, activity.ActivityTraceFlags, activity.TraceStateString, true))
```

Comment:
Previous logic was bugged. IsRemote was always false & SpanId was always different than the ParentSpanId that came in (by design).

Reply:
Fixed previously for AspNetCore on #1286.